swap_server_client.go

```diff
@@ -273,8 +299,109 @@ func (s *grpcSwapServerClient) NewLoopInSwap(ctx context.Context,
 	}, nil
 }
 
-func (s *grpcSwapServerClient) Close() {
-	s.conn.Close()
+// ServerUpdate summarizes an update from the swap server.
+type ServerUpdate struct {
+	// Hash is the hash that received the update.
+	Hash lntypes.Hash
+
+	// Update is the state that the server has sent us.
+	Update looprpc.ServerSwapState
+
+	// Timestamp is the time of the server update.
+	Timestamp time.Time
+}
+
+// LoopInState subscribes to loop in server state and pipes updates into the
+// channel provided.
+func (s *grpcSwapServerClient) LoopInState(ctx context.Context,
+	hash lntypes.Hash, updateChan chan<- *ServerUpdate) (<-chan error,
```

Comment:
in `lndclient` we generally use a pattern that returns both channels.

Reply:
We need to pass this in so that the daemon can consume from the updates channel (see next commit). If we don't do that, we need to return a channel here then fan into another channel; added complexity with not much benefit imo, and we already have this pattern with `statusChan`